### PR TITLE
Distill my current understanding of UPLC

### PR DESCRIPTION
**Pull request by srid**
More to come latter

### CODE REVIEW

File: uplc.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+     (DeBruijn {dbnIndex = 0})
+     (Var () (DeBruijn {dbnIndex = 1})))
+```
+
+## Pluto
+
+As you might have noticed it is not practical to write UPLC by hand. [Pluto](https://github.com/Plutonomicon/pluto) is a simple programming language that assembles directly to UPLC; that is to say, it 'maps' directly (more or less) to the AST of UPLC. Understanding Pluto facilitates an understanding of UPLC. Here's the equivalent Pluto program for the above UPLC example:
+
+```haskell
+(\× -> x)
```
Comment (ursi) [inline uplc.md]: weird `x` here

Comment (srid) [reply uplc.md]: Looks to be the Agda vscode extension getting in the way ...

File: uplc.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+## UPLC Representation
+
+Plutus represents a UPLC program using the following Haskell type:
+
+```haskell
+data Program name uni fun ann = 
+  Program ann (TPLC.Version ann) (Term name uni fun ann)
+```
+
+The part we are interested in is `Term name uni fun ann`. A UPLC is just a `Term` defined recursively in terms of itself, similar to languages like Lisp. The `Term` type is defined as follows:
```
Comment (ursi) [inline uplc.md]: is this supposed to say "A UPLC *program* is ..."?

**Review by morganthomas — APPROVED** [review]

File: uplc.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+```
+
+Bear in mind that UPLC is untyped, therefore these builtin functions will fail to evaluate if their types don't match what they expect.
+
+## UPLC values
+
+The *values* supported by UPLC are specified by the `DefaultUni` type.
+
+```haskell
+data DefaultUni a where
```
Comment (ursi) [inline uplc.md:83]: I think you should include these guys in this section as well https://github.com/input-output-hk/plutus/blob/e995df9a339b69523e34bad35816ee1e4ddd9669/plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs#L91-L94

**Pull request merged**